test/albumController.test.js

```diff
@@ -0,0 +1,105 @@
+const { app, connectDB } = require('../src/expressServer'); // Import your Express app here
+const { StatusCodes } = require('http-status-codes');
+const request = require('supertest');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const Album = require('../src/models/Album');
+
+const {
+  getAllAlbums,
+  getSingleAlbum,
+  getAlbumWithAllUsersWhoPurchasedIt,
+  getFilteredAlbums,
+} = require('../src/controllers/albumController');
+
+let server;
+let mongooseConnection;
+let mongodb;
+
+beforeAll(async () => {
+  mongodb = await MongoMemoryServer.create();
+  const url = mongodb.getUri();
+  process.env.MONGO_URL = url;
+  mongooseConnection = await connectDB(url);
+  server = await app.listen(8001);
+});
+
+afterAll(async () => {
+  await server.close();
+  await mongooseConnection.disconnect();
+  await mongodb.stop();
+});
+
+describe('AlbumController API Tests', () => {
+  const mockAlbumData = {
+    albumName: 'Test Album',
+    artistName: 'Test Artist',
+    price: 9.99,
+    spotifyUrl: 'https://api.spotify.com/v1/albums/',
```

Comment:
"Invalid Spotify URL" message comes from the mongoose validator for the album schema in the Album.js file on line 45 of that file: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/1e48ca722a1a852752cda040cae126beeffd535d/src/models/Album.js#L36-L47](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/1e48ca722a1a852752cda040cae126beeffd535d/src/models/Album.js#L36-L47)

The validator for the `spotifyUrl` field expects the following regex: `/^https:`**`\/\/`**`api\.spotify\.com`**`\/`**`v1`**`\/`**`albums`**`\/`**`[a`**`-`**`zA`**`-`**`Z0`**`-`**`9]+$/`

That regex means it's expecting a string with the shape: [`https://api.spotify.com/v1/albums/somelettersandornumbers`](https://api.spotify.com/v1/albums/somelettersandornumbers) but you are not passing in anything after the `/albums/` part of the URL so it is failing that check.

Reply:
Ok, should pass 'id' in some way or write this URL in another way?

Reply:
Since we're just creating a fake album for these tests, just putting any random characters after `/albums` should work. So something like:  
 

```suggestion
    spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
```

is fine

Reply:
Great , I didn't know it